picatrix/magics/timesketch.py

```diff
@@ -0,0 +1,1235 @@
+        'time_end': None,
+        'size': 10000,
+        'terminate_after': 10000,
+        'order': 'asc'
+    }
+
+  if indices:
+    query_filter['indices'] = indices
+  else:
+    query_filter['indices'] = '_all'
+
+  if start_date:
+    query_filter['time_start'] = start_date
+  if end_date:
+    query_filter['time_end'] = end_date
+
+  # If view is being sent in, view needs to be the only parameter to the search.
+  if view is not None:
+    query = None
+    query_dsl = None
+
+  return_fields = _fix_return_fields(return_fields)
+
+  return sketch.explore(
+      query_string=query, query_dsl=query_dsl,
+      query_filter=query_filter, view=view, return_fields=return_fields,
+      max_entries=max_entries, as_pandas=True)
+
+
+# pylint: disable=unused-argument
+@framework.picatrix_magic
+def timesketch_list_views(
+    data: Optional[Text] = '') -> Dict[str, api_view.View]:
+  """List up all available views.
+
+  Args:
+    data (str): Not used.
+
+  Returns:
+    A dict with a list of available views.
+  """
+  connect()
+  state_obj = state.state()
+  sketch = state_obj.get_from_cache('timesketch_sketch')
+
+  return_dict = {}
+  for view in sketch.list_views():
+    return_dict['{0:d}:{1:s}'.format(view.id, view.name)] = view
+  return return_dict
+
+
+@framework.picatrix_magic
+def timesketch_query(
+    data: Text,
+    fields: Optional[Text] = None,
+    timelines: Optional[Text] = None,
+    view: Optional[api_view.View] = None,
+    start_date: Optional[Text] = None,
+    end_date: Optional[Text] = None,
+    max_entries: Optional[int] = 40000) -> pd.DataFrame:
+  """Run a Timesketch query using a magic.
+
+  Args:
+    data (str): the Timesketch query.
+    fields (str): comma separated list of fields to include in the returned
+        data frame.
+    timelines (str): comma separated list of the names of the timelines.
+    view (View): the View object.
+    start_date (str): start date of the result set in the form of
+        YYYY-MM-DDTHH:MM:SS+00:00.
+    end_date (str): end date of the result set in the form of
+        YYYY-MM-DDTHH:MM:SS+00:00.
+    max_entries (int): maximum entries of returned object, defaults to 40k.
+
+  Raises:
+    ValueError: If a view object is passed in and it's not a proper view
+        object.
+
+  Returns:
+    DataFrame containing the results of the query.
+  """
+  connect()
+  state_obj = state.state()
+
+  if timelines:
+    sketch = state_obj.get_from_cache('timesketch_sketch')
+    timeline_list = sketch.list_timelines()
+    names = timelines.split(',')
+    indices = [
+        x.index for x in timeline_list if x.name in names]
+  else:
+    indices = None
+
+  if view and not is_view_object(view):
+    raise ValueError('View is not a view instance, but rather [{0:s}]'.format(
+        type(view)))
+
+  return query_timesketch(
+      query=data, view=view, return_fields=fields, indices=indices,
+      start_date=start_date, end_date=end_date, max_entries=max_entries)
```

Comment:
You want to move to `chip filters` here instead. The end_time and start_time is going to be deprecated at some point. I can fix this later though, so maybe just add a TODO?

Reply:
discussed offline.